### PR TITLE
Rename RepoHandler Functions

### PR DESCRIPTION
**Pull request by M-Davies**
* Updates the user scripts and defaults to account for the recent code split

Fixes: #34 
Signed-off-by: Morgan Davies <morgandavies2020@gmail.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
I realise now the PR tester will fail as it hasn't been merged yet 🤦 I'll try and figure out a way to test this alternatively

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -617,16 +619,8 @@ class Builder implements Serializable {
...
-            def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
-            context.println "Querying Adopt Api for the JDK-Head number (tip_version)..."
-
-            def response = JobHelper.getAvailableReleases(context)
-            int headVersion = (int) response.getAt("tip_version")
-            context.println "Found Java Version Number: ${headVersion}"
-    
-            if (javaToBuild == "jdk${headVersion}") {
+
+            if (javaToBuild == "jdk${getHeadVersionNumber()}") {
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:623]: Add comment to highlight what this special case is doing

**Comment by karianna** [comment]
@M-Davies compilation failure?

**Comment by M-Davies** [comment]
The `RepoHandlerTest` should be resolved upon merge. The `context.timeout(time: pipelineTimeouts.API_REQUEST_TIMEOUT, unit: "HOURS") {` are unrelated to this PR and generally caused by https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/issues/14

**Review by karianna — APPROVED** [review]

File: pipelines/library/src/common/RepoHandler.groovy [hunk 2]
```diff
@@ -18,51 +18,51 @@ class RepoHandler {
...
     public Map<String, ?> setUserDefaultsJson(String url) {
         def getUser = new URL(url).openConnection()
         this.USER_DEFAULTS_JSON = new JsonSlurper().parseText(getUser.getInputStream().getText()) as Map
     }
 
     /*
-    Changes dir to Adopt's repo
+    Changes dir to adopt's ci-jenkins-pipelines repo
     */
     public void checkoutAdopt () {
```
Comment (andrew-m-leonard) [inline pipelines/library/src/common/RepoHandler.groovy:57]: I think these 3 functions could be named better? say:
checkoutAdoptPipelines()
checkoutUserPipelines()
checkoutAdoptBuild()

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 3]
```diff
@@ -983,13 +983,13 @@ class Build {
...
-                                repoHandler.checkoutAdopt()
+                                context.println "[CHECKOUT] Checking out to AdoptOpenJDK/openjdk-build to use their shell scripts..."
+                                repoHandler.checkoutBuild()
                                 context.sh(script: "./build-farm/make-adopt-build-farm.sh")
                                 context.println "[CHECKOUT] Reverting pre-build AdoptOpenJDK/openjdk-build checkout..."
-                                context.checkout context.scm
+                                repoHandler.checkoutUser()
                             } else {
-                                context.println "[INFO] Executing user bash scripts..."
+                                context.println "[INFO] Executing user shell scripts..."
```
Comment (andrew-m-leonard) [inline pipelines/build/common/openjdk_build_pipeline.groovy:992]: Not sure the build-farm/make-adopt-build-farm.sh is going to be found as checkoutUser() checks out user pipelines repo, not a user build repo?

**Comment by M-Davies** [comment]
run tests

**Review by andrew-m-leonard — APPROVED** [review]
I think it's ok, although still a bit confused over checkouts... assuming tested ok, lets give it a try...

**Comment by M-Davies** [comment]
Probably a bit late to merge this today as I'm about to sign off. I'll look into merging tomorrow morning

**Comment by M-Davies** [comment]
Draft while ongoing external issues are being resolved

File: pipelines/library/src/common/RepoHandler.groovy [hunk 4]
```diff
@@ -18,57 +18,77 @@ class RepoHandler {
...
+        context.checkout([$class: 'GitSCM',
+            branches: [ [ name: ADOPT_DEFAULTS_JSON["repository"]["build_branch"] ] ],
+            userRemoteConfigs: [ [ url: ADOPT_DEFAULTS_JSON["repository"]["build_url"] ] ]
+        ])
+    }
+
+    /*
+    Changes dir to user's openjdk-build repo
+    */
+    public void checkoutUserBuild () {
```
Comment (M-Davies) [inline pipelines/library/src/common/RepoHandler.groovy:87]: TODO: Open an issue after merging to make this configurable by the constructor and class level

**Comment by M-Davies** [comment]
Merging and running a test pipeline and generation

**Pull request merged**